hydro_lang/src/location/mod.rs

```diff
@@ -127,10 +127,22 @@ pub trait Location<'a>: Clone {
 
     #[inline(never)]
     fn new_node_metadata<T>(&self) -> HydroIrMetadata {
```

Comment:
Who calls this API? We should remove it and rename the new one to `new_node_metadata`.